addons/website_sale/models/product_image.py

```diff
@@ -17,3 +20,62 @@ class ProductImage(models.Model):
+                                 'You can provide URL from Youtube, Instagram, Vimeo, Dailymotion or Youku.')
+    embed_code = fields.Char(compute="_compute_embed_code")
+
+    @api.depends('video_url')
+    def _compute_embed_code(self):
+        ''' Computes the valid iframe containing product video to be embedded
+            (or error message if URL is not valid).
+        '''
+
+        # To detect if we have a valid URL or not
+        urlRegex = r'^(http:\/\/|https:\/\/|\/\/)[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$'
+
+        # Regex for supported video types
+        ytRegex = r'^(?:(?:https?:)?\/\/)?(?:www\.)?(?:youtu\.be\/|youtube(-nocookie)?\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((?:\w|-){11})(?:\S+)?$'
+        vimeoRegex = r'\/\/(player.)?vimeo.com\/([a-z]*\/)*([0-9]{6,11})[?]?.*'
+        dmRegex = r'.+dailymotion.com\/(video|hub|embed)\/([^_]+)[^#]*(#video=([^_&]+))?'
+        igRegex = r'(.*)instagram.com\/p\/(.[a-zA-Z0-9]*)'
+        ykuRegex = r'(.*).youku\.com\/(v_show\/id_|embed\/)(.+)'
+
+        for image in self:
+            embedUrl = False
+            if image.video_url:
+                videoURL = image.video_url
+                validURL = re.search(urlRegex, image.video_url)
+
+                if not validURL:
+                    image.embed_code = _('Provided URL is not valid. Please enter a valid URL.')
+                else:
+                    ytMatch = re.search(ytRegex, videoURL)
+                    vimeoMatch = re.search(vimeoRegex, videoURL)
+                    dmMatch = re.search(dmRegex, videoURL)
+                    igMatch = re.search(igRegex, videoURL)
+                    ykuMatch = re.search(ykuRegex, videoURL)
+
+                    if ytMatch and len(ytMatch.groups()[1]) == 11:
+                        embedUrl = '//www.youtube%s.com/embed/%s?rel=0' % (ytMatch.groups()[0] or '', ytMatch.groups()[1])
+                    elif vimeoMatch:
+                        embedUrl = '//player.vimeo.com/video/%s' % (vimeoMatch.groups()[2])
+                    elif dmMatch:
+                        embedUrl = '//www.dailymotion.com/embed/video/%s' % (dmMatch.groups()[1])
+                    elif igMatch:
+                        embedUrl = '//www.instagram.com/p/%s/embed/' % (igMatch.groups()[1])
+                    elif ykuMatch:
+                        ykuLink = ykuMatch.groups()[2]
+                        if '.html?' in ykuLink:
+                            ykuLink = ykuLink.split('.html?')[0]
+                        embedUrl = '//player.youku.com/embed/%s' % (ykuLink)
+                    else:
+                        embedUrl = videoURL
+                    image.embed_code = '<iframe class="embed-responsive-item" src="%s" allowFullScreen="true" frameborder="0"></iframe>' % (embedUrl)
+
+    @api.constrains('video_url')
+    def _check_valid_video_url(self):
+        for image in self:
+            if image.video_url:
```

Comment:
if you have video_url, you enter line 46.
So embed_code was always set by L51 or L74 no?

Now if L51 we st to False, you need to check here '
if not image.embed_code or not ...'